internal/checker/checker.go

```diff
@@ -856,6 +856,8 @@ type Checker struct {
 	skipDirectInferenceNodes                    collections.Set[*ast.Node]
 	ctx                                         context.Context
 	packagesMap                                 map[string]bool
+	activeMappers                               []*TypeMapper
+	activeTypeMappersCaches                     []map[string]*Type
```

Comment:
The other "global" caches don't use two separate lists in the Go port, instead scoping them together, but I'm keeping them separate since I'm trying to reuse the maps.